src/Sylius/Bundle/CoreBundle/Doctrine/ORM/ProductRepository.php

```diff
@@ -92,6 +92,7 @@ public function createFilterPaginator($criteria = array(), $sorting = array(), $
 
         if ($deleted) {
             $this->_em->getFilters()->disable('softdeleteable');
+            $queryBuilder->andWhere('product.deletedAt IS NOT NULL');
```

Comment:
If you want only to list deleted products. There is no sense to list deleted and not deleted toghether